### PR TITLE
Remove use of satori/uuid package via upgrading azure-sdk-for-go

### LINKED ISSUE #1143: Update autorest and azure-sdk-for-go?
It looks like the autorest and sdk versions are quite old.  Could we get an update to pick up the recent 429 improvements?

### LINKED ISSUE #1291: Consider using a different UUID library
**Describe the current behavior**
We're using github.com/satori/go.uuid which hasnt been updated in a while.

**Describe the improvement**
Move to a different library. Overall usage is quite small.

### PR DESCRIPTION
**Pull request by Porges**
Closes #1291, closes #1143, closes #1306 (via obsoletion).

`go-autorest` also updated as `azure-sdk-for-go` required a newer version.

**What this PR does / why we need it**:

https://github.com/satori/go.uuid has an open security vulnerability. The new `azure-sdk-for-go` (as of v53) removes the dependency on `satori/go.uuid` (https://github.com/Azure/azure-sdk-for-go/issues/14283, https://github.com/Azure/azure-sdk-for-go/issues/3158).

Any direct references have been replaced by `gofrs/uuid`, which is API-compatible.

**How does this PR make you feel**:
gif

**If applicable**:
- [ ] this PR contains documentation
- [ ] this PR contains tests

### CODE REVIEW

**Review by theunrepentantgeek — APPROVED** [review]
:shipit:

**Review by babbageclunk — APPROVED** [review]

File: pkg/resourcemanager/eventhubs/namespace.go [hunk 1]
```diff
@@ -64,6 +64,14 @@ func (m *azureEventHubNamespaceManager) DeleteNamespace(ctx context.Context, res
 		resourceGroupName,
 		namespaceName)
 
+    if err != nil {
```
Comment (babbageclunk) [inline pkg/resourcemanager/eventhubs/namespace.go:67]: Is this just a drive-by?

Comment (Porges) [reply pkg/resourcemanager/eventhubs/namespace.go:67]: Failed here in CI…

Comment (matthchr) [reply pkg/resourcemanager/eventhubs/namespace.go:67]: Singe you asked offline, this looks "ok" to me (in that it's just doing what is also done above -- obviously the actual pattern is awkward but that's a problem for another day).

**Review by matthchr — APPROVED** [review]

**Comment by matthchr** [comment]
I wonder if we need to check our linter settings since this wasn't caught and is a lintable error, I think?
